src/linter/ui5Types/SourceFileLinter.ts

```diff
@@ -236,13 +224,26 @@ export default class SourceFileLinter {
 			])) {
 				return;
 			}
+
+			const nonStaticRender = (node as ts.ClassDeclaration).members.find((member: ts.ClassElement) => {
+				return (ts.isPropertyDeclaration(member) || ts.isMethodDeclaration(member)) &&
+					(ts.isIdentifier(member.name) || ts.isStringLiteral(member.name)) &&
+					member.name.text === "renderer";
+			});
+			if (nonStaticRender) {
```

Comment:
Shouldn't this check be done before the check for the special cases above:
```ts
this.isUi5ClassDeclaration(node, [
				"sap/ui/core/mvc/View",
				// XMLComposite is deprecated, but there still shouldn't be a false-positive about a missing renderer
				"sap/ui/core/XMLComposite",
				"sap/ui/core/webc/WebComponent",
				"sap/uxap/BlockBase",
			])
```
?

Otherwise a non static renderer of those subclasses won't be detected.

Reply:
Makes sense to move the before the spacial cases as it would never be able to detect that otherwise.